### ISSUE TITLE
[Bug]: Accordion icons do not update correctly in safari

### ISSUE THREAD

**Issue by emmayjiang** [microsoft/fluentui#24035]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Mac
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://fluentuipr.z22.web.core.windows.net/pull/24013/theme-designer/storybook/index.html?path=/story/themedesigner--page, viewed in safari on a Mac

## Actual Behavior
When the chevrons to open the accordions are clicked on, the content is displayed but the chevron does not update.
![image](https://user-images.githubusercontent.com/31319479/180490842-64726ac4-e949-4115-aad6-c8a14df571c7.png)

## Expected Behavior
The chevron should update to indicate when it is open and closed.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by JustSlone** [comment]
@microsoft/cxe-red, @tomi-msft we should understand the scope of this impact, if it really is affecting all icons across all components on safari this is medium or maybe even high. If this is just a corner case and affecting the theme designer this is low priority.

**Comment by spmonahan** [comment]
I believe this is the same issue I logged in #23163 and is affecting all icons in Safari (I only tested iPadOS but this issue calls out macOS).

The root of the issue appears to be that we're using a [transform to rotate the SVG](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx#L70). This also prevents custom expand icons from "just rotating" as the transform is only applied to the built in icon. My initial feeling was that this lack of rotation for custom icons is a bug but I could also see how the rotation is coupled with the icon being used so maybe this is fine. Regardless of where we land with custom icon rotation, the built in icons should definitely rotate.
